### PR TITLE
Add mount support

### PR DESCRIPTION
**Pull request by myusko**
#161

### CODE REVIEW

File: blacksheep/server/application.py [hunk 1]
```diff
@@ -629,7 +629,7 @@ def __init__(
         if mount_class is None:
             self._mount = Mount()
 
-    def mount(self, path: str, app):
+    def mount(self, path: str, app: Callable):
```
Comment (myusko) [inline blacksheep/server/application.py:632]: @RobertoPrevato Probably we need to create a custom ASGI annotation?

**Comment by myusko** [comment]
@RobertoPrevato PR is ready for review.

File: blacksheep/server/application.py [hunk 2]
```diff
@@ -608,3 +608,36 @@ async def __call__(self, scope, receive, send):
 
         request.scope = None  # type: ignore
         request.content.dispose()
+
+
+class ASGIApplication(Application):
```
Comment (myusko) [inline blacksheep/server/application.py:613]: Also, not sure that's the best name for the class.

File: blacksheep/server/application.py [hunk 3]
```diff
@@ -608,3 +608,36 @@ async def __call__(self, scope, receive, send):
...
+        mount_class: Optional[Mount] = None,
+    ):
+        super().__init__(
+            router=router,
+            services=services,
+            debug=debug,
+            show_error_details=show_error_details,
+        )
+        if mount_class is None:
+            self._mount = Mount()
```
Comment (RobertoPrevato) [inline blacksheep/server/application.py:630]: This is a mistake, the code is setting `self_mount` only if the input parameter is `None`.

You meant to do:

```python
if mount is None:
    mount = Mount()

self._mount: Mount = mount
```

File: blacksheep/server/application.py (hunk 2 above)
Comment (RobertoPrevato) [reply blacksheep/server/application.py:613]: I agree, I changed mind and don't like the name `ASGIApplication` - the `Application` class is already meant to be compatible with ASGI. Can you think of a better name? (I'll write something more in Gitter)

**Comment by RobertoPrevato** [comment]
Let's close this one, since I applied some changes on a new PR -> https://github.com/Neoteroi/BlackSheep/pull/177/files - where we result both contributors 😄

**Pull request closed without merging**